src/components/Card.js

```diff
@@ -39,21 +71,28 @@ class Card {
         this._handleCardImageClick({ name: this._name, link: this._link });
       });
 
-    this._cardElement
-      .querySelector(".card__delete-button")
-      .addEventListener("click", () => this._handleDeleteButton());
+    this.likeButton.addEventListener("click", () =>
+      this._handleLikeButton(this._id)
+    );
 
-    this._cardElement
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => this._handleLikeButton());
+    this.cardTrashButton.addEventListener("click", () => {
+      this._handleDeleteCard(this._id);
+    });
   }
 
   getCardElement() {
     this._cardElement = this._getTemplate();
-    this._setEventListeners();
     this._cardElement.querySelector(".card__title").textContent = this._name;
     this._cardElement.querySelector(".card__image").src = this._link;
```

Comment:
`this._cardElement.querySelector(".card__image")`

**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` and so on.